### PR TITLE
add explaining example to XLNet LM modeling

### LINKED ISSUE #2822: bugs in xlnet XLNetLMHeadModel
# 🐛 Bug

## Information

Model I am using  XLNet :

Language I am using the model on English :

The problem arises when using:
* [True ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. just deep into modeling_xlnet.py code;
2. browsing  https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_xlnet.py#L1057;
3. comparing with other LMHead,such as https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_gpt2.py#L603;
4.you'll found inputs and labels are not shifted.

## Expected behavior

the code should be refactored to the same style as other LMHeadModels
and because xlnet tokenizer appends <'sep'> , <'cls'> on the end, so the num tokens shifted should be 2, so the think the code should be: 
        if labels is not None:
            # Shift so that tokens < n predict n
            shift_logits = logits[..., :-2, :].contiguous()
            shift_labels = labels[..., 1:-1].contiguous()
            # Flatten the tokens
            loss_fct = CrossEntropyLoss()
            loss = loss_fct(logits.view(-1, logits.size(-1)), labels.view(-1))
            # loss = loss_fct(logits.view(-1, logits.size(-1)), labels.reshape(-1))
            outputs = (loss,) + outputs

## Environment info

     
- `transformers` version: 2.4.1
- Platform:
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):
- Using GPU in script?:
- Using distributed or parallel set-up in script?:

### PR DESCRIPTION
**Pull request by patrickvonplaten**
1) adds example explaining how XLNet could be used for standard auto-regressive modelling.
2) puts add_special_tokens=True for simple example to make sure no <sep> and <cls> are added to input. The are used for special training (similar to BERT) and might be confusing to user

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
Cool! Indeed, if I remember the paper correctly there is no need to shift the labels. Maybe we could update the [associated docstring](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_xlnet.py#L980-L985) to make sure no one gets confused again?

**Comment by patrickvonplaten** [comment]
> Cool! Indeed, if I remember the paper correctly there is no need to shift the labels. Maybe we could update the [associated docstring](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_xlnet.py#L980-L985) to make sure no one gets confused again?

Done :-)

**Review by LysandreJik — APPROVED** [review]
Great, thanks Patrick!
